src/data/nat/factorization/basic.lean

```diff
@@ -249,33 +249,86 @@ by { cases n, refl }
 lemma factorization_equiv_inv_apply {f : ℕ →₀ ℕ} (hf : ∀ p ∈ f.support, prime p) :
   (factorization_equiv.symm ⟨f, hf⟩).1 = f.prod pow := rfl
 
-/-! ### Factorization and divisibility -/
+/-! ### Generalisation of the "even part" and "odd part" of a natural number
 
-lemma dvd_of_mem_factorization {n p : ℕ} (h : p ∈ n.factorization.support) : p ∣ n :=
-begin
-  rcases eq_or_ne n 0 with rfl | hn, { simp },
-  simp [←mem_factors_iff_dvd hn (prime_of_mem_factorization h), factor_iff_mem_factorization.mp h],
-end
+We introduce the notations `ord_proj[p] n` for the largest power of the prime `p` that
+divides `n` and `ord_compl[p] n` for the complementary part. The `ord` naming comes from
+the $p$-adic order/valuation of a number, and `proj` and `compl` are for the projection and
+complementary projection. The term `n.factorization p` is the $p$-adic order itself.
+For example, `ord_proj[2] n` is the even part of `n` and `ord_compl[2] n` is the odd part. -/
 
-lemma pow_factorization_dvd (n p : ℕ) : p ^ n.factorization p ∣ n :=
+notation `ord_proj[` p `]` n:max := p ^ (nat.factorization n p)
+notation `ord_compl[` p `]` n:max := n / ord_proj[p] n
+
+@[simp] lemma ord_proj_of_not_prime (n p : ℕ) (hp : ¬ p.prime) : ord_proj[p] n = 1 :=
+by simp [factorization_eq_zero_of_non_prime n hp]
+
+@[simp] lemma ord_compl_of_not_prime (n p : ℕ) (hp : ¬ p.prime) : ord_compl[p] n = n :=
+by simp [factorization_eq_zero_of_non_prime n hp]
+
+lemma ord_proj_dvd (n p : ℕ) : ord_proj[p] n ∣ n :=
 begin
-  by_cases hp : p.prime, swap, { simp [factorization_eq_zero_of_non_prime n hp] },
-  rw ←factors_count_eq,
+  by_cases hp : p.prime, swap, { simp [hp] },
+  rw [←factors_count_eq],
```

Comment:
I don't really see the need for this change but I don't want to block the whole PR on it either. Still I prefer:
```suggestion
  rw ←factors_count_eq,
```